core/net/src/hybrid/cli.rs

```diff
@@ -85,3 +97,74 @@ fn to_status_metrics(metrics: &mut ChannelMetrics) -> model::StatusMetrics {
         rx_current: metrics.rx.short.average(time),
     }
 }
+
+pub async fn cli_ping(_msg: model::GsbPing) -> Result<Vec<GsbPingResponse>, StatusError> {
+    let client = {
+        CLIENT.with(|c| c.borrow().clone()).ok_or_else(|| {
+            model::StatusError::RuntimeException("client not initialized".to_string())
+        })?
+    };
+
+    // This will update sessions ping. We don't display them in this view
+    // but I think it is good place to enforce this.
+    client.ping_sessions().await;
+
+    let nodes = client.connected_nodes().await;
+    let our_node_id = client.node_id();
+    let ping_timeout = Duration::from_secs(10);
+
+    log::debug!("Ping: Num connected nodes: {}", nodes.len());
+
+    let results = join_all(
+        nodes
+            .iter()
+            .map(|(id, alias)| async move {
+                let tcp_before = Instant::now();
+
+                ya_net::from(our_node_id)
+                    .to(*id)
+                    .service(ya_net::DIAGNOSTIC)
+                    .send(GsbRemotePing {})
+                    .timeout(Some(ping_timeout))
+                    .await???;
+
+                let tcp_ping = tcp_before.elapsed();
+                let udp_before = Instant::now();
+
+                ya_net::from(our_node_id)
+                    .to(*id)
+                    .service_udp(ya_net::DIAGNOSTIC)
+                    .send(GsbRemotePing {})
+                    .timeout(Some(ping_timeout))
+                    .await???;
+
+                let udp_ping = udp_before.elapsed();
+
+                anyhow::Ok(GsbPingResponse {
+                    node_id: *id,
+                    node_alias: alias.clone(),
+                    tcp_ping,
+                    udp_ping,
+                })
+            })
+            .map(|future| future.map_err(|e| StatusError::RuntimeException(e.to_string())))
```

Comment:
It may be useful to know which flavor of ping failed - whether it was udp or a virtual tcp connection. Pings would have to be reordered though